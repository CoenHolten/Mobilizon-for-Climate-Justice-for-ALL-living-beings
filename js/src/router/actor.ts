import Profile from '@/views/Account/Profile.vue';
import CreateGroup from '@/views/Group/Create.vue';
import Group from '@/views/Group/Group.vue';
import GroupList from '@/views/Group/GroupList.vue';
import Identities from '@/views/Account/Identities.vue';
import { RouteConfig } from 'vue-router';

export enum ActorRouteName {
  IDENTITIES = 'Identities',
  GROUP_LIST = 'GroupList',
  GROUP = 'Group',
  CREATE_GROUP = 'CreateGroup',
  PROFILE = 'Profile',
}

export const actorRoutes: RouteConfig[] = [
  {
    path: '/identities',
    name: ActorRouteName.IDENTITIES,
    component: Identities,
    meta: { requiredAuth: true },
  },
  {
    path: '/groups',
    name: ActorRouteName.GROUP_LIST,
    component: GroupList,
    meta: { requiredAuth: false },
  },
  {
    path: '/groups/create',
    name: ActorRouteName.CREATE_GROUP,
    component: CreateGroup,
    meta: { requiredAuth: true },
  },
  {
    path: '/~:preferredUsername',
    name: ActorRouteName.GROUP,
    component: Group,
    props: true,
    meta: { requiredAuth: false },
  },
  {
    path: '/@:name',
    name: ActorRouteName.PROFILE,
    component: Profile,
    props: true,
    meta: { requiredAuth: false },
  },
];
