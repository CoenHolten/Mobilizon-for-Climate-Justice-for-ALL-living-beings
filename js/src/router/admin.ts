import { RouteConfig } from 'vue-router';
import Dashboard from '@/views/Admin/Dashboard.vue';
import Follows from '@/views/Admin/Follows.vue';
import Followings from '@/components/Admin/Followings.vue';
import Followers from '@/components/Admin/Followers.vue';

export enum AdminRouteName {
  DASHBOARD = 'Dashboard',
  RELAYS = 'Relays',
  RELAY_FOLLOWINGS = 'Followings',
  RELAY_FOLLOWERS = 'Followers',
}

export const adminRoutes: RouteConfig[] = [
  {
    path: '/admin',
    name: AdminRouteName.DASHBOARD,
    component: Dashboard,
    props: true,
    meta: { requiredAuth: true },
  },
  {
    path: '/admin/relays',
    name: AdminRouteName.RELAYS,
    redirect: { name: AdminRouteName.RELAY_FOLLOWINGS },
    component: Follows,
    children: [
      {
        path: 'followings',
        name: AdminRouteName.RELAY_FOLLOWINGS,
        component: Followings,
      },
      {
        path: 'followers',
        name: AdminRouteName.RELAY_FOLLOWERS,
        component: Followers,
      },
    ],
    props: true,
    meta: { requiredAuth: true },
  },
];
