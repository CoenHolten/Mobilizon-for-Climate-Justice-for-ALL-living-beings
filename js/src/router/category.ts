import CategoryList from '@/views/Category/List.vue';
import CreateCategory from '@/views/Category/Create.vue';

export enum CategoryRouteName {
  CATEGORY_LIST = 'CategoryList',
  CREATE_CATEGORY = 'CreateCategory',
}

export const categoryRoutes = [
  {
    path: '/category',
    name: CategoryRouteName.CATEGORY_LIST,
    component: CategoryList,
    meta: { requiredAuth: false },
  },
  {
    path: '/category/create',
    name: CategoryRouteName.CREATE_CATEGORY,
    component: CreateCategory,
    meta: { requiredAuth: true },
  },
];
