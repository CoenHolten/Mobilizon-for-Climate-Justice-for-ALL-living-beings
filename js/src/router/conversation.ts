import { RouteConfig } from "vue-router";
import CreateConversation from "@/views/Conversations/Create.vue";
import ConversationsList from "@/views/Conversations/ConversationsList.vue";
import Conversation from "@/views/Conversations/Conversation.vue";

export enum ConversationRouteName {
  CONVERSATION_LIST = "CONVERSATION_LIST",
  CREATE_CONVERSATION = "CREATE_CONVERSATION",
  CONVERSATION = "CONVERSATION",
}

export const conversationRoutes: RouteConfig[] = [
  {
    path: "/@:preferredUsername/conversations",
    name: ConversationRouteName.CONVERSATION_LIST,
    component: ConversationsList,
    props: true,
    meta: { requiredAuth: false },
  },
  {
    path: "/@:preferredUsername/conversations/new",
    name: ConversationRouteName.CREATE_CONVERSATION,
    component: CreateConversation,
    props: true,
    meta: { requiredAuth: true },
  },
  {
    path: "/@:preferredUsername/:slug/:id/:comment_id?",
    name: ConversationRouteName.CONVERSATION,
    component: Conversation,
    props: true,
    meta: { requiredAuth: false },
  },
];
