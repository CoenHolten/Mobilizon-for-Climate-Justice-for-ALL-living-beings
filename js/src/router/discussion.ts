import { RouteConfig } from "vue-router";

export enum DiscussionRouteName {
  DISCUSSION_LIST = "DISCUSSION_LIST",
  CREATE_DISCUSSION = "CREATE_DISCUSSION",
  DISCUSSION = "DISCUSSION",
}

export const discussionRoutes: RouteConfig[] = [
  {
    path: "/@:preferredUsername/discussions",
    name: DiscussionRouteName.DISCUSSION_LIST,
    component: () =>
      import(/* webpackChunkName: "DiscussionsList" */ "@/views/Discussions/DiscussionsList.vue"),
    props: true,
    meta: { requiredAuth: false },
  },
  {
    path: "/@:preferredUsername/discussions/new",
    name: DiscussionRouteName.CREATE_DISCUSSION,
    component: () =>
      import(/* webpackChunkName: "CreateDiscussion" */ "@/views/Discussions/Create.vue"),
    props: true,
    meta: { requiredAuth: true },
  },
  {
    path: "/@:preferredUsername/c/:slug/:comment_id?",
    name: DiscussionRouteName.DISCUSSION,
    component: () =>
      import(/* webpackChunkName: "Discussion" */ "@/views/Discussions/Discussion.vue"),
    props: true,
    meta: { requiredAuth: false },
  },
];
