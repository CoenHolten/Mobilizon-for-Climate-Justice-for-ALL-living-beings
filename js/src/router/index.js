import Vue from 'vue';
import Router from 'vue-router';
import PageNotFound from '@/components/PageNotFound';
import Home from '@/components/Home';
import Event from '@/components/Event/Event';
import EventList from '@/components/Event/EventList';
import Location from '@/components/Location';
import CreateEvent from '@/components/Event/Create';
import CategoryList from '@/components/Category/List';
import CreateCategory from '@/components/Category/Create';
import Register from '@/components/Account/Register';
import Login from '@/components/Account/Login';
import Validate from '@/components/Account/Validate';
import ResendConfirmation from '@/components/Account/ResendConfirmation';
import SendPasswordReset from '@/components/Account/SendPasswordReset';
import PasswordReset from '@/components/Account/PasswordReset';
import Account from '@/components/Account/Account';
import CreateGroup from '@/components/Group/Create';
import Group from '@/components/Group/Group';
import GroupList from '@/components/Group/GroupList';
import Identities from '../components/Account/Identities.vue';

Vue.use(Router);

const router = new Router({
  mode: 'history',
  base: '/',
  routes: [
    {
      path: '/',
      name: 'Home',
      component: Home,
      meta: { requiredAuth: false },
    },
    {
      path: '/events/list/:location?',
      name: 'EventList',
      component: EventList,
      meta: { requiredAuth: false },
    },
    {
      path: '/events/create',
      name: 'CreateEvent',
      component: CreateEvent,
      meta: { requiredAuth: true },
    },
    {
      path: '/events/:id(\\d+)/edit',
      name: 'EditEvent',
      component: CreateEvent,
      props: true,
      meta: { requiredAuth: true },
    },
    {
      path: '/location/new',
      name: 'Location',
      component: Location,
      meta: { requiredAuth: true },
    },
    {
      path: '/category',
      name: 'CategoryList',
      component: CategoryList,
      meta: { requiredAuth: false },
    },
    {
      path: '/category/create',
      name: 'CreateCategory',
      component: CreateCategory,
      meta: { requiredAuth: true },
    },
    {
      path: '/register',
      name: 'Register',
      component: Register,
      props: true,
      meta: { requiredAuth: false },
    },
    {
      path: '/resend-instructions',
      name: 'ResendConfirmation',
      component: ResendConfirmation,
      props: true,
      meta: { requiresAuth: false },
    },
    {
      path: '/password-reset/send',
      name: 'SendPasswordReset',
      component: SendPasswordReset,
      props: true,
      meta: { requiresAuth: false },
    },
    {
      path: '/password-reset/:token',
      name: 'PasswordReset',
      component: PasswordReset,
      meta: { requiresAuth: false },
      props: true,
    },
    {
      path: '/validate/:token',
      name: 'Validate',
      component: Validate,
      props: true,
      meta: { requiresAuth: false },
    },
    {
      path: '/login',
      name: 'Login',
      component: Login,
      props: true,
      meta: { requiredAuth: false },
    },
    {
      path: '/identities',
      name: 'Identities',
      component: Identities,
      meta: { requiredAuth: true },
    },
    {
      path: '/groups',
      name: 'GroupList',
      component: GroupList,
      meta: { requiredAuth: false },
    },
    {
      path: '/group-create',
      name: 'CreateGroup',
      component: CreateGroup,
      meta: { requiredAuth: true },
    },
    {
      path: '/~:name',
      name: 'Group',
      component: Group,
      props: true,
      meta: { requiredAuth: false },
    },
    {
      path: '/@:name',
      name: 'Account',
      component: Account,
      props: true,
      meta: { requiredAuth: false },
    },
    {
      path: '/events/:uuid',
      name: 'Event',
      component: Event,
      props: true,
      meta: { requiredAuth: false },
    },
    {
      path: '*',
      name: 'PageNotFound',
      component: PageNotFound,
      meta: { requiredAuth: false },
    },
  ],
});

export default router;
