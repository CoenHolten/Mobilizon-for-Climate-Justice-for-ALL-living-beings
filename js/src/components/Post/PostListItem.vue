<template>
  <router-link
    class="block md:flex bg-white dark:bg-violet-2 dark:text-white dark:hover:text-white rounded-lg shadow-md"
    dir="auto"
    :to="{ name: RouteName.POST, params: { slug: post.slug } }"
  >
    <lazy-image-wrapper
      :picture="post.picture"
      :rounded="true"
      class="object-cover flex-none h-48 md:h-auto md:w-48 rounded-t-lg md:rounded-none md:rounded-l-lg"
    />
    <div class="flex flex-col gap-1 bg-inherit p-2 rounded-lg flex-1">
      <h3
        class="text-xl color-violet-3 line-clamp-3 mb-2 font-bold"
        :lang="post.language"
      >
        {{ post.title }}
      </h3>
      <p class="flex gap-2">
        <Clock />
        <span dir="auto" class="" v-if="isBeforeLastWeek">{{
          formatDateTimeString(
            publishedAt.toString(),
            undefined,
            false,
            "short"
          )
        }}</span>
        <span v-else>{{
          formatDistanceToNow(publishedAt, {
            locale: dateFnsLocale,
            addSuffix: true,
          })
        }}</span>
      </p>
      <div v-if="postTags.length > 0" class="flex flex-wrap gap-y-0 gap-x-2">
        <Tag />
        <mbz-tag v-for="tag in postTags" :key="tag.slug">{{
          tag.title
        }}</mbz-tag>
      </div>
      <p class="flex gap-1" v-if="isCurrentActorMember">
        <AccountEdit />
        <i18n-t keypath="Published by {name}">
          <template #name>
            <b class="">{{ displayName(post.author) }}</b>
          </template>
        </i18n-t>
      </p>
    </div>
  </router-link>
</template>
<script lang="ts" setup>
import { formatDistanceToNow, subWeeks, isBefore, Locale } from "date-fns";
import RouteName from "@/router/name";
import { IPost } from "@/types/post.model";
import LazyImageWrapper from "@/components/Image/LazyImageWrapper.vue";
import { displayName } from "@/types/actor";
import { computed, inject } from "vue";
import { formatDateTimeString } from "@/filters/datetime";
import Tag from "vue-material-design-icons/Tag.vue";
import AccountEdit from "vue-material-design-icons/AccountEdit.vue";
import Clock from "vue-material-design-icons/Clock.vue";
import MbzTag from "@/components/TagElement.vue";

const props = withDefaults(
  defineProps<{
    post: IPost;
    isCurrentActorMember?: boolean;
  }>(),
  { isCurrentActorMember: false }
);

const dateFnsLocale = inject<Locale>("dateFnsLocale");

const postTags = computed(() => (props.post.tags ?? []).slice(0, 3));

const publishedAt = computed((): Date => {
  return new Date((props.post.publishAt ?? props.post.insertedAt) as Date);
});

const isBeforeLastWeek = computed((): boolean => {
  return isBefore(publishedAt.value, subWeeks(new Date(), 1));
});
</script>
<style lang="scss" scoped>
@use "@/styles/_mixins" as *;

// .post-minimalist-card-wrapper {
//   display: grid;
//   grid-gap: 5px 10px;
//   grid-template-areas: "preview" "body";
//   text-decoration: none;
//   // width: 100%;
//   // color: initial;

//   // @include desktop {
//   grid-template-columns: 200px 3fr;
//   grid-template-areas: "preview body";
//   // }

.title-info-wrapper {
  .post-minimalist-title {
    font-size: 18px;
    line-height: 24px;
    font-weight: 700;
    overflow: hidden;
    display: -webkit-box;
    -webkit-box-orient: vertical;
    -webkit-line-clamp: 3;
  }
}
:deep(.icon) {
  vertical-align: middle;
  // @include margin-right(5px);
}

:deep(.tags) {
  display: inline;

  span.tag {
    max-width: 200px;

    & > span {
      overflow: hidden;
      text-overflow: ellipsis;
    }
  }
}
// }
</style>
