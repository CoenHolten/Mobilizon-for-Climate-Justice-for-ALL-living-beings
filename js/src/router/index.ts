import Vue from 'vue';
import Router from 'vue-router';
import PageNotFound from '@/views/PageNotFound.vue';
import Home from '@/views/Home.vue';
import { UserRouteName, userRoutes } from './user';
import { EventRouteName, eventRoutes } from '@/router/event';
import { ActorRouteName, actorRoutes } from '@/router/actor';
import { CategoryRouteName, categoryRoutes } from '@/router/category';

Vue.use(Router);

enum GlobalRouteName {
  HOME = 'Home',
  PAGE_NOT_FOUND = 'PageNotFound',
}

// Hack to merge enums
// tslint:disable:variable-name
export const RouteName = {
  ...GlobalRouteName,
  ...UserRouteName,
  ...EventRouteName,
  ...CategoryRouteName,
  ...ActorRouteName,
};

const router = new Router({
  mode: 'history',
  base: '/',
  routes: [
    ...userRoutes,
    ...eventRoutes,
    ...categoryRoutes,
    ...actorRoutes,

    {
      path: '/',
      name: RouteName.HOME,
      component: Home,
      meta: { requiredAuth: false },
    },

    {
      path: '*',
      name: RouteName.PAGE_NOT_FOUND,
      component: PageNotFound,
      meta: { requiredAuth: false },
    },
  ],
});

export default router;
