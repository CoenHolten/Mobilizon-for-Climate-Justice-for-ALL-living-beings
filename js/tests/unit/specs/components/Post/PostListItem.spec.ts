import { config, mount } from "@vue/test-utils";
import PostListItem from "@/components/Post/PostListItem.vue";
import { vi, beforeEach, describe, it, expect } from "vitest";
import { enUS } from "date-fns/locale";
import { routes } from "@/router";
import { createRouter, createWebHistory, Router } from "vue-router";

let router: Router;

beforeEach(async () => {
  router = createRouter({
    history: createWebHistory(),
    routes: routes,
  });

  // await router.isReady();
});

const postData = {
  id: "1",
  slug: "my-blog-post-some-uuid",
  title: "My Blog Post",
  body: "My content",
  insertedAt: "2020-12-02T09:01:20.873Z",
  tags: [],
  language: "en",
};

const generateWrapper = (
  customPostData: Record<string, unknown> = {},
  customProps: Record<string, unknown> = {}
) => {
  return mount(PostListItem, {
    props: {
      post: { ...postData, ...customPostData },
      ...customProps,
    },
    global: {
      provide: {
        dateFnsLocale: enUS,
      },
      plugins: [router],
    },
  });
};

describe("PostListItem", () => {
  it("renders post list item with basic informations", () => {
    const wrapper = generateWrapper();

    expect(wrapper.html()).toMatchSnapshot();

    expect(wrapper.find("a.block.bg-white").attributes("href")).toBe(
      `/p/${postData.slug}`
    );

    expect(wrapper.find("h3").text()).toBe(postData.title);

    expect(wrapper.find("p.flex.gap-2").text()).toBe("Dec 2, 2020");

    expect(wrapper.find("p.flex.gap-1").exists()).toBeFalsy();
  });

  it("renders post list item with tags", () => {
    const wrapper = generateWrapper({
      tags: [{ slug: "a-tag", title: "A tag" }],
    });

    expect(wrapper.html()).toMatchSnapshot();

    expect(wrapper.find("div.flex.flex-wrap.gap-y-0.gap-x-2").text()).toContain(
      "A tag"
    );

    expect(wrapper.find("p.flex.gap-1").exists()).toBeFalsy();
  });

  it("renders post list item with publisher name", () => {
    const wrapper = generateWrapper(
      { author: { name: "An author" } },
      { isCurrentActorMember: true }
    );

    expect(wrapper.html()).toMatchSnapshot();

    expect(wrapper.find("p.flex.gap-1").exists()).toBeTruthy();
    expect(wrapper.find("p.flex.gap-1").text()).toContain("An author");
  });
});
