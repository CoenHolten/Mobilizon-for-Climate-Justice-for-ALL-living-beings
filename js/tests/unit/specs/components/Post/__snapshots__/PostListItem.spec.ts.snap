// Vitest Snapshot v1

exports[`PostListItem > renders post list item with basic informations 1`] = `
"<a href=\\"/p/my-blog-post-some-uuid\\" class=\\"block md:flex bg-white dark:bg-violet-2 dark:text-white dark:hover:text-white rounded-lg shadow-md\\" dir=\\"auto\\" data-v-6ca7cc69=\\"\\">
  <!--v-if-->
  <div class=\\"flex flex-col gap-1 bg-inherit p-2 rounded-lg flex-1\\" data-v-6ca7cc69=\\"\\">
    <h3 class=\\"text-xl color-violet-3 line-clamp-3 mb-2 font-bold\\" lang=\\"en\\" data-v-6ca7cc69=\\"\\">My Blog Post</h3>
    <p class=\\"flex gap-2\\" data-v-6ca7cc69=\\"\\"><span aria-hidden=\\"true\\" class=\\"material-design-icon clock-icon\\" role=\\"img\\" data-v-6ca7cc69=\\"\\"><svg fill=\\"currentColor\\" class=\\"material-design-icon__svg\\" width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\"><path d=\\"M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2M16.2,16.2L11,13V7H12.5V12.2L17,14.9L16.2,16.2Z\\"><!--v-if--></path></svg></span><span dir=\\"auto\\" class=\\"\\" data-v-6ca7cc69=\\"\\">Dec 2, 2020</span></p>
    <!--v-if-->
    <!--v-if-->
  </div>
</a>"
`;

exports[`PostListItem > renders post list item with publisher name 1`] = `
"<a href=\\"/p/my-blog-post-some-uuid\\" class=\\"block md:flex bg-white dark:bg-violet-2 dark:text-white dark:hover:text-white rounded-lg shadow-md\\" dir=\\"auto\\" data-v-6ca7cc69=\\"\\">
  <!--v-if-->
  <div class=\\"flex flex-col gap-1 bg-inherit p-2 rounded-lg flex-1\\" data-v-6ca7cc69=\\"\\">
    <h3 class=\\"text-xl color-violet-3 line-clamp-3 mb-2 font-bold\\" lang=\\"en\\" data-v-6ca7cc69=\\"\\">My Blog Post</h3>
    <p class=\\"flex gap-2\\" data-v-6ca7cc69=\\"\\"><span aria-hidden=\\"true\\" class=\\"material-design-icon clock-icon\\" role=\\"img\\" data-v-6ca7cc69=\\"\\"><svg fill=\\"currentColor\\" class=\\"material-design-icon__svg\\" width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\"><path d=\\"M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2M16.2,16.2L11,13V7H12.5V12.2L17,14.9L16.2,16.2Z\\"><!--v-if--></path></svg></span><span dir=\\"auto\\" class=\\"\\" data-v-6ca7cc69=\\"\\">Dec 2, 2020</span></p>
    <!--v-if-->
    <p class=\\"flex gap-1\\" data-v-6ca7cc69=\\"\\"><span aria-hidden=\\"true\\" class=\\"material-design-icon account-edit-icon\\" role=\\"img\\" data-v-6ca7cc69=\\"\\"><svg fill=\\"currentColor\\" class=\\"material-design-icon__svg\\" width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\"><path d=\\"M21.7,13.35L20.7,14.35L18.65,12.3L19.65,11.3C19.86,11.09 20.21,11.09 20.42,11.3L21.7,12.58C21.91,12.79 21.91,13.14 21.7,13.35M12,18.94L18.06,12.88L20.11,14.93L14.06,21H12V18.94M12,14C7.58,14 4,15.79 4,18V20H10V18.11L14,14.11C13.34,14.03 12.67,14 12,14M12,4A4,4 0 0,0 8,8A4,4 0 0,0 12,12A4,4 0 0,0 16,8A4,4 0 0,0 12,4Z\\"><!--v-if--></path></svg></span>Published by <b class=\\"\\" data-v-6ca7cc69=\\"\\">An author</b></p>
  </div>
</a>"
`;

exports[`PostListItem > renders post list item with tags 1`] = `
"<a href=\\"/p/my-blog-post-some-uuid\\" class=\\"block md:flex bg-white dark:bg-violet-2 dark:text-white dark:hover:text-white rounded-lg shadow-md\\" dir=\\"auto\\" data-v-6ca7cc69=\\"\\">
  <!--v-if-->
  <div class=\\"flex flex-col gap-1 bg-inherit p-2 rounded-lg flex-1\\" data-v-6ca7cc69=\\"\\">
    <h3 class=\\"text-xl color-violet-3 line-clamp-3 mb-2 font-bold\\" lang=\\"en\\" data-v-6ca7cc69=\\"\\">My Blog Post</h3>
    <p class=\\"flex gap-2\\" data-v-6ca7cc69=\\"\\"><span aria-hidden=\\"true\\" class=\\"material-design-icon clock-icon\\" role=\\"img\\" data-v-6ca7cc69=\\"\\"><svg fill=\\"currentColor\\" class=\\"material-design-icon__svg\\" width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\"><path d=\\"M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2M16.2,16.2L11,13V7H12.5V12.2L17,14.9L16.2,16.2Z\\"><!--v-if--></path></svg></span><span dir=\\"auto\\" class=\\"\\" data-v-6ca7cc69=\\"\\">Dec 2, 2020</span></p>
    <div class=\\"flex flex-wrap gap-y-0 gap-x-2\\" data-v-6ca7cc69=\\"\\"><span aria-hidden=\\"true\\" class=\\"material-design-icon tag-icon\\" role=\\"img\\" data-v-6ca7cc69=\\"\\"><svg fill=\\"currentColor\\" class=\\"material-design-icon__svg\\" width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\"><path d=\\"M5.5,7A1.5,1.5 0 0,1 4,5.5A1.5,1.5 0 0,1 5.5,4A1.5,1.5 0 0,1 7,5.5A1.5,1.5 0 0,1 5.5,7M21.41,11.58L12.41,2.58C12.05,2.22 11.55,2 11,2H4C2.89,2 2,2.89 2,4V11C2,11.55 2.22,12.05 2.59,12.41L11.58,21.41C11.95,21.77 12.45,22 13,22C13.55,22 14.05,21.77 14.41,21.41L21.41,14.41C21.78,14.05 22,13.55 22,13C22,12.44 21.77,11.94 21.41,11.58Z\\"><!--v-if--></path></svg></span><span class=\\"rounded-md my-1 truncate text-sm text-violet-title px-2 py-1 bg-purple-3 dark:text-violet-3\\" data-v-6955ca87=\\"\\" data-v-6ca7cc69=\\"\\">A tag</span></div>
    <!--v-if-->
  </div>
</a>"
`;
